### DISCUSSION TITLE
Ant runinstall failed with : CLI could not contact EJBCA instance.

### DISCUSSION THREAD

**Discussion by gilles380** [Keyfactor/ejbca-ce#533] [Q&A] [answered]
Hi support team,
I traveled all along the install procedure (Mariadb/WildFly,and ejbca github ) , then if fail on last mile, I read in the forum about wildfly possibly not well configured, but as I am not skilled in application server, I don't have any clue.
My installation 
EJBCA from github https://github.com/Keyfactor/ejbca-ce (ant deployear, ant build then ant runinstall) 
WildFly is 26 (following https://doc.primekey.com/ejbca/ejbca-installation/application-servers/jboss-eap-7-4#JBossEAP7.4-ConfigureWildFlyRemoting)
MariaDB 11.3.2, database setup (https://doc.primekey.com/ejbca/ejbca-installation/creating-the-database#CreatingtheDatabase-MariaDB/MySQL)
Please find the WildFly log attached
[wildfly.log](https://github.com/Keyfactor/ejbca-ce/files/14514290/wildfly.log)
Then I can provide other usefull information for you to help me on this.
Many thanks 
G.

**Comment by svenska-primekey** [comment]
From the log there is no ejbca.ear file being loaded. What is in the deployment directory for wildfly? If you followed the directions the path should be /opt/wildfly/standalone/deployment

**Reply by gilles380** [reply]
Hi @svenska-primekey thanks for your quick reply ! 
you're right, because of long install travell I did not notice about the fact that 'ant deployear' did not copy the ear file dut to premissions directory issue.
I correct the issue, now I have 

~/ejbca-ce$ ls -ail /opt/wildfly/standalone/deployments/

50593806 -rw-rw-r-- 1 wildfly wildfly 43711268 mars   7 09:54 ejbca.ear
50593808 -rw-r--r-- 1 wildfly wildfly      418 mars   7 09:54 ejbca.ear.failed
50602395 -rw-r--r-- 1 wildfly wildfly   624101 sept. 25  2020 mariadb-java-client.jar
50593797 -rw-r--r-- 1 wildfly wildfly       23 sept. 25  2020 mariadb-java-client.jar.deployed

but still, runinstall failed with same error, may be mariadb database  issue ? 
ejbca:init:
     [echo] 
     [echo] ------------------- CA Properties ----------------
     [echo] ca.name                : ManagementCA
     [echo] ca.dn                  : CN=ManagementCA,O=EJBCA Sample,C=SE
     [echo] ca.tokentype           : soft
     [echo] ca.keytype             : RSA
     [echo] ca.keyspec             : 2048
     [echo] ca.signaturealgorithm  : SHA256WithRSA
     [echo] ca.validity            : 3650
     [echo] ca.policy              : null
     [echo] ca.tokenproperties     : ${ca.tokenproperties}
     [echo] httpsserver.hostname   : localhost
     [echo] httpsserver.dn         : CN=localhost,O=EJBCA Sample,C=SE
     [echo] httpsserver.tokentype  : JKS
     [echo] superadmin.cn          : SuperAdmin
     [echo] superadmin.dn          : CN=SuperAdmin
     [echo] superadmin.batch       : true
     [echo] appserver.home         : /opt/wildfly-26.0.0.Final
     [echo]         

ejbca:install:

ejbca:initCA:
     [echo] Initializing CA with 'ManagementCA' 'CN=ManagementCA,O=EJBCA Sample,C=SE' 'soft' '<ca.tokenpassword hidden>' '2048' 'RSA' '3650' 'null' 'SHA256WithRSA'   -superadmincn 'SuperAdmin'...
     [java] 09:55:47.522 [main] ERROR org.ejbca.ui.cli.infrastructure.library.CommandLibrary - Error: CLI could not contact EJBCA instance. Either your application server is not up and running, EJBCA has not been deployed successfully, or some firewall rule is blocking the CLI from the application server.
[java] Exception in thread "main" javax.ejb.NoSuchEJBException: EJBCLIENT000079: Unable to discover destination for request for EJB StatelessEJBLocator for "ejbca/cesecore-ejb/GlobalConfigurationSessionBean", view is interface org.cesecore.configuration.GlobalConfigurationSessionRemote, affinity is None

+
sudo systemctl status wildfly
● wildfly.service - The WildFly Application Server
     Loaded: loaded (/etc/systemd/system/wildfly.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2024-03-06 19:47:00 CET; 14h ago
   Main PID: 600 (launch.sh)
      Tasks: 122 (limit: 37985)
     Memory: 853.5M
        CPU: 1min 1.469s
     CGroup: /system.slice/wildfly.service
             ├─600 /bin/bash /opt/wildfly/bin/launch.sh standalone standalone.xml 0.0.0.0
             ├─633 /bin/sh /opt/wildfly/bin/standalone.sh -c standalone.xml -b 0.0.0.0
             └─915 java "-D[Standalone]" -server -Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs= -Djava.awt.headless=true --a>
mars 06 19:47:00 NUC systemd[1]: Started The WildFly Application Server._

+
/usr/local/mysql$ sudo bin/mysqld_safe --user=mysql
[sudo] Mot de passe de matter : 
bin/mysqld_safe: Deprecated program name. It will be removed in a future release, use 'mariadbd-safe' instead
240306 19:48:17 mysqld_safe Logging to '/usr/local/mariadb-11.3.2-linux-systemd-x86_64/data/matter-NUC.err'.
240306 19:48:17 mysqld_safe Starting mariadbd daemon with databases from /usr/local/mariadb-11.3.2-linux-systemd-x86_64/data

**Reply by primetomas** [reply]
Now there is likely some error occuring while deploying, such as problems connecting to your database (the most common issue).
This will be visible in the server.log of WildFly.
https://doc.primekey.com/ejbca/troubleshooting-guide

**Reply by gilles380** [reply]
here is the log after I restarted wildfly service
[wildfly.log](https://github.com/Keyfactor/ejbca-ce/files/14521999/wildfly.log)

**Reply by gilles380** [reply]
Also, I have

2024-03-07 10:41:36,448 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("deploy") failed - address: ([("deployment" => "ejbca.ear")]) - failure description: {
    "WFLYCTL0412: Required services that are not installed:" => ["jboss.naming.context.java.EjbcaDS"],
    "WFLYCTL0180: Services with missing/unavailable dependencies" => [
        "jboss.persistenceunit.\"ejbca.ear#ejbca\".__FIRST_PHASE__ is missing [jboss.naming.context.java.EjbcaDS]",
        "jboss.persistenceunit.\"ejbca.ear#ejbca\" is missing [jboss.naming.context.java.EjbcaDS]"
    ]
}
2024-03-07 10:41:36,514 ERROR [org.jboss.as] (Controller Boot Thread) WFLYSRV0026: WildFly Full 26.0.0.Final (WildFly Core 18.0.0.Final) started (with errors) in 7607ms - Started 5377 of 6249 services (674 services failed or missing dependencies, 409 services are lazy, passive or on-demand)

**Reply by primetomas** [reply]
"missing [jboss.naming.context.java.EjbcaDS]" looks like either this command wasn't executed, or didn't succeed, i.e. adding the data source (EjbcaDS) to WildFly configuration.
https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26#WildFly26-AddDatabaseDriver
and
https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26#WildFly26-AddaDatasource

**Reply by primetomas** [reply]
You missed to set MySQL as database in:
https://doc.primekey.com/ejbca/ejbca-installation/managing-ejbca-configurations#ManagingEJBCAConfigurations-database.properties

**Reply by primetomas** [reply, marked as answer]
"Unknown table 'sequences' in information_schema" (most likely) means the information in database.properties was wrong when the ear file was built. If you change values afterwards you must:
"ant clean deployear"

Cheers

**Reply by gilles380** [reply]
Thanks @primetomas  and @svenska-primekey  for your support.
